### PR TITLE
gh-2992 Ensure ExecutionContext is properly deserialized

### PR DESCRIPTION
**Pull request by ghillert**
* Add tests
* Refactor and unify Jackson MixIns
* Ensure that the Jackson dependencies are the same for Server and Rest Client

resolves https://github.com/spring-cloud/spring-cloud-dataflow/issues/2992

### CODE REVIEW

File: spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/DataflowTemplateTests.java [hunk 1]
```diff
@@ -86,6 +86,25 @@ public void testDataFlowTemplateContructorWithNonExistingUri() throws URISyntaxE
...
+	@Test
+	public void testPrepareObjectMapperWithNullObjectMapper() {
+		try {
+			DataFlowTemplate.prepareObjectMapper(null);
+		}
+		catch (IllegalArgumentException e) {
+			assertEquals("The objectMapper must not be null.", e.getMessage());
+			return;
+		}
+		fail("Expected an IllegalArgumentException to be thrown.");
```
Comment (cppwfs) [inline spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/DataflowTemplateTests.java]: We can move the fail inside the try block.  To be consistent with the other changes.

File: spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/JobExecutionDeserializationTests.java [hunk 2]
```diff
@@ -56,62 +42,49 @@
...
-		Assert.assertEquals("Expect 1 JobExecutionInfoResource", 6, paged.getContent().size());
-		Assert.assertEquals(Long.valueOf(6), jobExecutionResource.getJobId());
-		Assert.assertEquals("job200616815", jobExecutionResource.getName());
-		Assert.assertEquals("COMPLETED", jobExecutionResource.getJobExecution().getStatus().name());
-
+		assertEquals("Expect 1 JobExecutionInfoResource", 6, paged.getContent().size());
+		assertEquals(Long.valueOf(6), jobExecutionResource.getJobId());
+		assertEquals("job200616815", jobExecutionResource.getName());
+		assertEquals("COMPLETED", jobExecutionResource.getJobExecution().getStatus().name());
+		assertEquals("COMPLETED", jobExecutionResource.getJobExecution().getStatus().name());
```
Comment (cppwfs) [inline spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/JobExecutionDeserializationTests.java]: Duplicate test.

File: spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/JobExecutionDeserializationTests.java [hunk 3]
```diff
@@ -56,62 +42,49 @@
...
+		assertNotNull(jobExecutionInfoResource);
+		assertEquals(Long.valueOf(1), jobExecutionInfoResource.getJobId());
+		assertEquals("ff.job", jobExecutionInfoResource.getName());
+		assertEquals("COMPLETED", jobExecutionInfoResource.getJobExecution().getStatus().name());
+		assertEquals(1, jobExecutionInfoResource.getJobExecution().getStepExecutions().size());
+
+		StepExecution stepExecution = jobExecutionInfoResource.getJobExecution().getStepExecutions().iterator().next();
+		assertNotNull(stepExecution);
+
+		ExecutionContext stepExecutionExecutionContext = stepExecution.getExecutionContext();
```
Comment (cppwfs) [inline spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/JobExecutionDeserializationTests.java]: add final to be consistent

File: spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/JobExecutionDeserializationTests.java [hunk 4]
```diff
@@ -56,62 +42,49 @@
...
-		Assert.assertEquals(Long.valueOf(1), jobExecutionInfoResource.getJobId());
-		Assert.assertEquals("ff.job", jobExecutionInfoResource.getName());
-		Assert.assertEquals("COMPLETED", jobExecutionInfoResource.getJobExecution().getStatus().name());
+		assertNotNull(jobExecutionInfoResource);
+		assertEquals(Long.valueOf(1), jobExecutionInfoResource.getJobId());
+		assertEquals("ff.job", jobExecutionInfoResource.getName());
+		assertEquals("COMPLETED", jobExecutionInfoResource.getJobExecution().getStatus().name());
+		assertEquals(1, jobExecutionInfoResource.getJobExecution().getStepExecutions().size());
+
+		StepExecution stepExecution = jobExecutionInfoResource.getJobExecution().getStepExecutions().iterator().next();
```
Comment (cppwfs) [inline spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/JobExecutionDeserializationTests.java]: add final to be consistent

**Comment by ghillert** [comment]
@cppwfs Rebased + addressed all requested changes.

**Comment by cppwfs** [comment]
LGTM

**Comment by cppwfs** [comment]
Squashed, Rebased, merged

**Pull request closed without merging**